t/t3428-rebase-edit-todo.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+test_description='rebase should reread the todo file if an exec modifies it'
+
+. ./test-lib.sh
+
+todo=$(git rev-parse --git-dir)/rebase-merge/git-rebase-todo
```

Comment:
> `foo=$(echo look  at    these       white spaces); echo "$foo"`

That's not the lack of quotes around the variable assignment, though. The extra whitespace is between arguments to `echo`. So `echo` sees it turned into a list and puts a single space between each word. You can see the same effect without a variable assignment:

```
$ echo look  at    these       white spaces
look at these white spaces
```